packages/server-core/src/user/user/user.hooks.ts

```diff
@@ -314,6 +285,26 @@ const handleUserSearch = async (context: HookContext<UserService>) => {
   }
 }
 
+const addLastLogin = async (context: HookContext<UserService>) => {
+  if (!context.result) return
+
+  const results = (
+    Array.isArray(context.result) ? context.result : 'data' in context.result ? context.result.data : [context.result]
+  ) as UserType[]
+
+  for (const item of results) {
+    const user = item as UserType
+    const lastLogin = await context.app.service(userLoginPath).find({
+      query: {
+        userId: user.id,
+        $sort: { createdAt: -1 },
+        $limit: 1
+      }
+    })
+    user.lastLogin = lastLogin.data[0]
+  }
+}
```

Comment:
Why is this removed from resolver?

Reply:
Because it should only be resolved for external requests with admin scope

Reply:
This can go in external resolver and in resolver we can put a check for scopes.

In current implementation `iff(isProvider('external'), verifyScope('admin', 'admin'), addLastLogin)` will cause all external find calls to have `admin:admin` scope. Is it really intended? because if verifyScope hook will throw exeception if that particular scope is missing. For checking, checkScope hook is preferred as it will return true, false. i.e. https://github.com/ir-engine/ir-engine/blob/9895e970b442607a5d77499831cb28ccad293215/packages/server-core/src/setting/project-setting/project-setting.hooks.ts#L85

Reply:
Hmm @barankyle ?

Reply:
I agree with Hanzla, doing this verifyScope on admin:admin would throw errors as-constructed if a non-admin with user:read ran user.FIND. I'm ambivalent on doing checkScope in the external resolver or as part of an iffElse in the hooks.